src/make.ts

```diff
@@ -446,17 +446,6 @@ export async function generateParseContent(progress: vscode.Progress<{}>,
     // Continue with the make dryrun invocation
     let makeArgs: string[] = [];
 
-    // Prepend the target to the arguments given in the makefile.configurations object,
-    // unless we want to parse for the full set of available targets.
-    if (forTargets) {
-        makeArgs.push("all");
-    } else {
-        let currentTarget: string | undefined = configuration.getCurrentTarget();
-        if (currentTarget) {
-            makeArgs.push(currentTarget);
-        }
-    }
-
     // Include all the make arguments defined in makefile.configurations.makeArgs
     makeArgs = makeArgs.concat(configuration.getConfigurationMakeArgs());
```

Comment:
I see why you moved makeArgs.push(currentAll) to later below... but what about "all" in case of "forTargets"? You did not do that intentionally? When "forTargets" is true, we need a slightly different set of args passed to "make" and if not correct we don't identify correctly all available targets. We need "all" in that case and with your change I don't see where we make sure we add "all".

Reply:
Recall that in #217, "all" does not work for target discovery.

Removing "all" as an explicit target means make will work on the .DEFAULT_GOAL, whatever that happens to be (should be "all" in most or all autotools projects, but it can be something else, like "sub-make" in #217

This won't fix 217, because in that example the wrapper makefile has no ability to recurse, but it will avoid an unnecesary error ("target 'all' does not exist" in cases like #217). Adding 'all' to this command does nothing, as far as I can see.

Was "all" added for some specific purpose, where the implicit target was not better?